### PR TITLE
CRAW-206 | Robots txt ep

### PR DESCRIPTION
**Pull request by mante-hike**
https://hikeseo.atlassian.net/browse/CRAW-206

### CODE REVIEW

File: app/Http/Controllers/Api/RobotsController.php [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+namespace App\Http\Controllers\Api;
+
+use App\Http\Controllers\Controller;
+use App\Http\Requests\RobotsRequest;
+use Illuminate\Http\JsonResponse;
+use Illuminate\Support\Facades\Http;
+
+class RobotsController extends Controller
+{
+    public function __invoke(RobotsRequest $request): JsonResponse
```
Comment (Dan-Hike) [inline app/Http/Controllers/Api/RobotsController.php:12]: In the both the v3 and v4 crawler projects, it looks like we handle redirects and follow them, presumably incase `/robots.txt` is redirected to the actual source. Probably worth replicating this in ours
https://github.com/Hike-SEO/CrawlerAPI/blob/main/src/robots/index.ts
https://github.com/kieranheadley/NodeCrawler/blob/master/functions/src/api/robots/getRobots.ts

File: app/Http/Controllers/Api/RobotsController.php [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    {
+        $robotsUrl = parse_url($request->websiteUrl, PHP_URL_SCHEME)
+            .'://'.parse_url($request->websiteUrl, PHP_URL_HOST);
+
+        return response()->json([
+            Http::timeout(3)
+                ->withOptions(['allow_redirects' => ['max' => 2]])
+                ->get("{$robotsUrl}/robots.txt")
+                ->body(),
+        ]);
```
Comment (ross-hike) [inline app/Http/Controllers/Api/RobotsController.php:22]: Can we move this logic into a service please?

File: tests/TestCase.php [hunk 3]
```diff
@@ -7,4 +7,13 @@
...
     use CreatesApplication;
+
+    public function usingTestApiToken(): static
+    {
+        config()->set([
+            'crawler.api_key' => 'secret-key',
+        ]);
+
+        return $this->withHeader('x-api-key', 'secret-key');
+    }
```
Comment (ross-hike) [inline tests/TestCase.php:18]: Noice

**Review by ross-hike — CHANGES_REQUESTED** [review]
Nice one, just a small request plz mate

**Comment by mante-hike** [comment]
@ross-hike nice one mate

**Review by james-hike — APPROVED** [review]

**Review by ross-hike — APPROVED** [review]

**Pull request merged**